engines/batch-ensembler/ensembler/__init__.py

```diff
@@ -0,0 +1,59 @@
+from typing import DefaultDict
+import yaml
+from pyspark import SparkConf, SparkContext
+from pyspark.sql import SparkSession
+from ensembler.job import BatchEnsemblingJob
+from ensembler.ensembler import Ensembler
+
+
+def build_spark_session(
+        app_name: str,
+        spark_config: DefaultDict[str, str] = None,
+        hadoop_config: DefaultDict[str, str] = None) -> SparkSession:
+    conf = SparkConf()
+    if spark_config:
+        conf.setAll(spark_config.items())
+
+    sc = SparkContext(conf=conf)
+
+    if hadoop_config:
+        for k, v in hadoop_config.items():
+            sc._jsc.hadoopConfiguration().set(k, v)
+
+    return SparkSession.builder \
+        .appName(app_name) \
+        .config(conf=sc.getConf()) \
+        .getOrCreate()
+
+
+class SparkApplication(object):
+    _ANNOTATION_GROUP_SPARK = 'spark'
+    _ANNOTATION_GROUP_HADOOP = 'hadoopConfiguration'
+
+    def __init__(self, args):
+        self.job, raw_config = BatchEnsemblingJob.from_yaml(args.job_spec)
+        print(
+            'Job Configuration: \n'
+            f'{yaml.dump(raw_config)}'
+        )
+
+    def run(self):
+        annotations = {}
+        for name, value in self.job.annotations().items():
```

Comment:
In this case parsing an annotation without `/` in it will throw a runtime error. (If annotation name without `/` is a legit one is another question)

```python
group, *keys = "io.k8s".split('/', 1)
# group == 'io.k8s'
# keys == []

group, keys = "io.k8s".split('/', 1)
# ValueError: not enough values to unpack (expected 2, got 1)
```

Reply:
Ohh okay :)